daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/Bytes.scala

```diff
@@ -51,14 +50,43 @@ object Bytes {
     new Bytes(value)
 
   def fromByteArray(a: Array[Byte]): Bytes =
-    new Bytes(ByteString.copyFrom(a))
+    fromByteArray(a, 0, a.length)
+
+  def fromByteArray(a: Array[Byte], offset: Int, size: Int) =
+    new Bytes(ByteString.copyFrom(a, offset, size))
 
   def fromByteBuffer(a: ByteBuffer): Bytes =
     new Bytes(ByteString.copyFrom(a))
 
   def fromInputStream(a: InputStream): Bytes =
     new Bytes(ByteString.readFrom(a))
 
+  // reads at most `len` bytes, does not close `a`
+  def fromInputStream(a: InputStream, len: Int): Bytes = {
```

Comment:
this seems like a fair amount of complexity and potential source of bugs. do the performance gains matter?

Reply:
I reverted the optimization. Do you prefer like that.

Reply:
thx seems simpler! And just to be clear, I’m happy to make an optimization here if it provides a benefit just not without some measurements.

Reply:
:+1: